app/models/reaction.rb

```diff
@@ -114,6 +114,23 @@ def user_has_been_given_too_many_spammy_comment_reactions?(user:, threshold: 2,
     def user_has_spammy_profile_reaction?(user:)
       user_vomits.exists?(reactable_id: user.id)
     end
+
+    # @param category [String] the reaction category type, see the CATEGORIES var
+    # @param reactable_id [Boolean] the ID of the item that was reacted on
+    # @param reactable_type [String] the type of the item, see the REACTABLE_TYPES var
+    # @param user [User] a moderator user
+
+    # @return [Array] Reactions that contain a contradictory category to the category that was passed in,
+    # example, if we pass in a "thumbsup", then we return reactions that have have a thumbsdown or vomit
+    def contradictory_mod_reactions(category:, reactable_id:, reactable_type:, user:)
+      if category == "thumbsup"
```

Comment:
This is pretty great! Should it maybe be a class-level method? Or, if it's at the object level, should it use the object's data (for example, for `category`) ?

Also, one thing that's more visible at this point in the refactoring is that the branches of this `if/else` mostly differ in how they determine what `category` is `contradictory`. It sort of makes the other differences looks like distractions. What about something like this:

```ruby
    contradictory_category = NEGATIVE_PRIVILEGED_CATEGORIES if category == "thumbsup"
    contradictory_category = "thumbsup" if category.in?(NEGATIVE_PRIVILEGED_CATEGORIES)
    
    where(reactable_id: reactable_id, 
          reactable_type: reactable_type, 
          user: current_user,
          category: contradictory_category)
```